x-pack/plugins/observability_solution/observability_ai_assistant/public/hooks/use_genai_connectors.ts

```diff
@@ -22,70 +24,61 @@ export interface UseGenAIConnectorsResult {
 
 export function useGenAIConnectors(): UseGenAIConnectorsResult {
   const assistant = useObservabilityAIAssistant();
+  const { services } = useKibana();
 
-  return useGenAIConnectorsWithoutContext(assistant);
+  return useGenAIConnectorsWithoutContext(assistant, services);
+}
+
+export function fetchConnectors(assistant: ObservabilityAIAssistantService) {
+  return assistant.callApi('GET /internal/observability_ai_assistant/connectors', {
+    signal: null,
+  });
 }
 
 export function useGenAIConnectorsWithoutContext(
-  assistant: ObservabilityAIAssistantService
+  assistant: ObservabilityAIAssistantService,
+  coreStart: CoreStart
 ): UseGenAIConnectorsResult {
+  const { uiSettings } = coreStart;
   const [connectors, setConnectors] = useState<FindActionResult[] | undefined>(undefined);
 
-  const [selectedConnector, setSelectedConnector] = useLocalStorage(
```

Comment:
`savedConnector` is the (saved) default connector, and `selectedConnector` is the selected one within the given session. We no longer have a concept of "last used" that persists across reloads. Do we want that? Should changing the connector from within the chat also persist that change to ES?

Reply:
maybe this one should be in localStorage? this sounds like it will change quite often depending on what the user is doing, and it probably should also be user specific?

Reply:
I tend to agree. But that opens a lot of questions. It both has technical implications and UX implications. Should we still show all settings in the same UI? How do users distinguish between those stored in ES and those stored in local storage?